includes/wizards/class-advertising-wizard.php

```diff
@@ -222,22 +203,24 @@ public function register_api_endpoints() {
 			[
 				'methods'             => \WP_REST_Server::EDITABLE,
 				'callback'            => [ $this, 'api_update_adunit' ],
-				'permission_callback' => [ $this, 'api_permissions_check_unfiltered_html' ],
+				'permission_callback' => [ $this, 'api_permissions_check' ],
 				'args'                => [
-					'id'          => [
+					'id'         => [
 						'sanitize_callback' => 'absint',
 					],
-					'name'        => [
+					'name'       => [
+						'sanitize_callback' => 'sanitize_text_field',
+					],
+					'code'       => [
 						'sanitize_callback' => 'sanitize_text_field',
 						'validate_callback' => [ $this, 'api_validate_not_empty' ],
 					],
-					'ad_code'     => [
-						// 'sanitize_callback' => 'esc_js', @todo If a `script` tag goes here, esc_js is the wrong function to use.
+					'sizes'      => [
+						// 'sanitize_callback' => 'sanitize_text_field', TODO: Correct sanitization for this field.
```

Comment:
Can we sanitize to make sure it's an array where each element is numeric or `''`? Does that meet all the use-cases?

Reply:
Added sanitization in https://github.com/Automattic/newspack-plugin/pull/375/commits/2009e4e7e49be060444557367edb86fb7abb4712. It's identical to sanitization being performed in Newspack Ads, but there's certainly no harm in doing it twice.